ui/app/controllers/jobs/index.js

```diff
@@ -194,6 +199,29 @@ export default class IndexController extends Controller.extend(
     return availableNamespaces;
   }
 
+  @computed('selectionNodePools', 'model.nodePools.[]')
+  get optionsNodePools() {
+    const availableNodePools = this.model.nodePools;
+
+    scheduleOnce('actions', () => {
+      // eslint-disable-next-line ember/no-side-effects
+      this.set(
+        'qpNodePool',
+        serialize(
+          intersection(
+            availableNodePools.map(({ name }) => name),
+            this.selectionNodePools
+          )
+        )
+      );
+    });
```

Comment:
Would this be better off run from a method elsewhere, or in a constructor?

Reply:
A `constructor` isn't the appropriate place to run this callback function because it runs whenever the controller is instantiated and there is a change to the dependent keys listed. Therefore it is best to keep the functionality here. I don't think its worth engineering effort until we upgrade to a `tracked` paradigm for the controller.